provisioning/miniprov/hare_mp/utils.py

```diff
@@ -349,6 +349,20 @@ def save_ssl_config(self):
         })
         self.kv.kv_put('ssl/hax', ssl_hax)
 
+    '''
+    Provisioner will be generating node_group from each data pod and place
+    gconf copy into consul. Hence from consul for all data pods node_group
+    value will be available.
+    '''
+    @func_log(func_enter, func_leave)
+    @repeat_if_fails()
+    def get_node_group(self, machine_id: str):
+        key = f'conf/node>{machine_id}>node_group'
+        node_group = self.kv.kv_get(key)
```

Comment:
Do we want to be backward compatible? what if node group is not configured and corresponding kv is not present. `kv_get()` will raise HAConsistency exception.

Reply:
@mssawant  Added backward compatibility.
For backward compatibility we will check if node_group info is present for current node, if it is present then we will wait for other node else we will assume that node_group concept is not supported.